azure-pipelines-e2e-integration-tests.yml

```diff
@@ -5,124 +5,154 @@
-  displayName: 'Build & Run tests for java 11'
-  continueOnError: false
+    - pwsh: |
+        java -version
+      displayName: 'Check default java version'
+
+    - pwsh: |
+        Invoke-WebRequest 'https://raw.githubusercontent.com/dotnet/cli/master/scripts/obtain/dotnet-install.ps1' -OutFile 'dotnet-install.ps1'
+            ./dotnet-install.ps1 -InstallDir "$env:ProgramFiles/dotnet" -Version "6.0.100-rc.2.21505.57" -Channel 'release'
+      displayName: 'Install the .Net version used by the Core Tools for Windows'
+      condition: eq( variables['Agent.OS'], 'Windows_NT' )
+    - bash: |
+        curl -sSL https://raw.githubusercontent.com/dotnet/cli/master/scripts/obtain/dotnet-install.sh | bash /dev/stdin -v '6.0.100-rc.2.21505.57' -c 'release' --install-dir /usr/share/dotnet
+      displayName: 'Install the .Net version used by the Core Tools for Linux'
+      condition: eq( variables['Agent.OS'], 'Linux' )
+
+    - pwsh: | # Download JDK for later installation 
+        Invoke-WebRequest $(JDK_DOWNLOAD_LINK) -O "$(JAVA_VERSION).tar.gz"
+        $current = get-location | select -ExpandProperty Path
+        Write-Host "##vso[task.setvariable variable=downloadPath;]$current"
+      displayName: 'Download jdk for Linux'
+      condition: eq( variables['Agent.OS'], 'Linux' )
+    - task: JavaToolInstaller@0 # Install JDK downloaded from previous task
+      inputs:
+        versionSpec: $(JAVA_VERSION_SPEC)
+        jdkArchitectureOption: 'x64'
+        jdkSourceOption: LocalDirectory
+        jdkFile: "$(downloadPath)/$(JAVA_VERSION).tar.gz"
+        jdkDestinationDirectory: "$(downloadPath)/externals"
+        cleanDestinationDirectory: true
+      condition: eq( variables['Agent.OS'], 'Linux' )
+      displayName: 'Setup Java for Linux'
+    - pwsh: |
+        Invoke-WebRequest $(JDK_DOWNLOAD_LINK) -O "$(JAVA_VERSION).zip"
+        Expand-Archive -Force "$(JAVA_VERSION).zip" .
+        cd $(JDK_PATH)
+        $current = get-location | select -ExpandProperty Path
+        cd ..
+        Write-Host "##vso[task.setvariable variable=JavaHome;]$current"
+      displayName: 'Download and setup Java for Windows'
+      condition: eq( variables['Agent.OS'], 'Windows_NT' )
+
+    - pwsh: |
+          .\setup-tests-pipeline.ps1 -UseCoreToolsBuildFromIntegrationTests
+      displayName: 'Setup test environment -- Install the Core Tools'
+    - pwsh: |
+        $currDir =  Get-Location
+        $Env:Path = $Env:Path+";$currDir\Azure.Functions.Cli"
+        ls $currDir\Azure.Functions.Cli
+        func --version
+      condition: eq( variables['Agent.OS'], 'Windows_NT' )
+      displayName: 'Setup Core Tools - Windows'
+    - bash: |
+        chmod +x ./Azure.Functions.Cli/func
+        chmod +x ./Azure.Functions.Cli/gozip
+        ls ./Azure.Functions.Cli
+        export PATH=$PATH:./Azure.Functions.Cli
+        func --version
+      condition: eq( variables['Agent.OS'], 'Linux' )
+      displayName: 'Setup Core Tools - Linux'
+
+    - pwsh: |
```

Comment:
I think we should move the jdk install logics after this task, we want to make sure `mvn clean package` runs on java8 only.